### PR TITLE
Safer shimming

### PR DESCRIPTION
**Pull request by bengl**
### What does this PR do?
Adds a layer of try/catch around instrumentation code, and makes a best-effort attempt to attribute errors to instrumentation (i.e. rather than the instrumented code). In those cases, it handles the errors transparently. This is all only enabled in SSI.

### Additional Notes

Caveats about code that isn't covered here are mentioned in the code comments.

### CODE REVIEW

**Comment by simon-id** [comment]
I don't have time to review today but what kind of error would it be able to handle ?

**Comment by bengl** [comment]
@simon-id this is specifically covering errors thrown in instrumentation code on the publisher side of the diagnostics channel.

File: packages/datadog-shimmer/test/shimmer.spec.js [hunk 1]
```diff
@@ -227,6 +227,207 @@ describe('shimmer', () => {
...
+                throw new Error('wrapper error')
+              }
+              return returnValue
+            }
+          })
+
+          expect(await obj.count()).to.equal(3)
+        })
+      })
+      // describe('callback', () => {
```
Comment (Qard) [inline packages/datadog-shimmer/test/shimmer.spec.js:397]: Maybe add a comment to explain why all this is here but commented out.

**Review by Qard — APPROVED** [review]
LGTM, just one suggestion.

**Comment by bengl** [comment]
@Qard I converted this back to draft because I'm concerned that `shimmer.wrap()` may have been broken by this approach, when SSI is enabled. Until I verify that's all good, I'd prefer to leave this in draft mode.

**Review by Qard — DISMISSED** [review]

**Review by Qard — DISMISSED** [review]

**Review by Qard — APPROVED** [review]

**Pull request merged**